### DISCUSSION TITLE
"Could not prepare Boot variable: Invalid argument"

### DISCUSSION THREAD

**Discussion by almightiest** [Sithuk/ubuntu-server-zfsbootmenu#47] [General]
My noble/24.04 server encrypted mirror install is failing at the end of step 4 with an error, and I could use help on how to recover/proceed:

`...
Running in chroot, ignoring request.
+ sh /etc/zfsbootmenu/generate-zbm.post.d/esp-sync.sh
     78,037,802 100%  767.24MB/s    0:00:00 (xfr#97, to-chk=0/104) 
0
1
Could not prepare Boot variable: Invalid argument
ubuntu@ubuntu:~$ 
`

I am using the latest version of the script and configured it.  Then I ran the install and selected my 2 mirror drives by id.  The rest ran auto until it failed on the update_boot_manager() function.

if i run the `efibootmgr` command in line 1461 of the script directly I can reproduce this issue.

`efibootmgr --create --disk /dev/nvme2n1 --label "rEFInd Boot Manager Backup 1" --loader \\EFI\\refind\\refind_x64.efi`

returns "Could not prepare Boot variable: Invalid argument"

**Comment by almightiest** [comment]
I resolved this by going into my bios and loading factory default secure boot keys. Dunno why that worked, but that's all I did to correct it after hours of searching and tinkering.

**Reply by Sithuk** [reply]
Thank you for the feedback.
